ziti/cmd/verify/ops_verify_network.go

```diff
@@ -0,0 +1,263 @@
+				log.Infof("Verifying controller config: %s", n.controllerConfig)
+				anyFailure = verifyControllerConfig(n.controllerConfig) || anyFailure
+				fmt.Println()
+			}
+			if n.routerConfig != "" {
+				log.Infof("Verifying router config: %s", n.routerConfig)
+				anyFailure = verifyRouterConfig(n.routerConfig) || anyFailure
+				fmt.Println()
+			}
+			if anyFailure {
+				log.Error("One or more error. Review the output above for errors.")
+			} else {
+				log.Info("All requested checks passed.")
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&n.controllerConfig, "controller-config-file", "c", "", "the controller config file verify")
+	cmd.Flags().StringVarP(&n.routerConfig, "router-config-file", "r", "", "the router config file to verify")
+	cmd.Flags().BoolVar(&n.verbose, "verbose", false, "Show additional output.")
+
+	return cmd
+}
+
+type StringMap map[string]interface{}
+
+func (m StringMap) mapFromKey(key string) StringMap {
+	if v, ok := m[key]; ok {
+		return v.(StringMap)
+	}
+	log.Fatalf("map didn't contain key %s", key)
+	return nil
+}
+
+type StringMapList []interface{}
+
+func (m StringMap) listFromKey(key string) StringMapList {
+	if v, ok := m[key]; ok {
+		return v.([]interface{})
+	}
+	log.Fatalf("map didn't contain key %s", key)
+	return nil
+}
+
+func (p ProtoHostPort) testPort(msg string) bool {
+	conn, err := net.DialTimeout("tcp", p.address(), 3*time.Second)
+	if err != nil {
+		log.Errorf("%s at %s cannot be reached.", msg, p.address())
+		return true
+	}
+	defer func(conn net.Conn) {
+		_ = conn.Close()
+	}(conn)
+	log.Infof("%s at %s is available.", msg, p.address())
+	return false
+}
+
+type ProtoHostPort struct {
+	proto string
+	host  string
+	port  string
+}
+
+func (p ProtoHostPort) address() string {
+	return p.host + ":" + p.port
+}
+
+func fromString(input string) *ProtoHostPort {
+	// input is expected to be in either "proto:host:port" format or "host:port" format
+	r := new(ProtoHostPort)
+	parts := strings.Split(input, ":")
+	if len(parts) > 2 {
+		r.proto = parts[0]
+		r.host = parts[1]
+		r.port = parts[2]
+	} else if len(parts) > 1 {
+		r.proto = "none"
+		r.host = parts[0]
+		r.port = parts[1]
+	} else {
+		panic("input is invalid: " + input)
+	}
+	return r
+}
+
+func verifyControllerConfig(controllerConfigFile string) bool {
+	if _, err := os.Stat(controllerConfigFile); err != nil {
+		log.Errorf("controller config file %s does not exist", controllerConfigFile)
+		return true
+	}
+	ctrlCfgBytes, err := os.ReadFile(controllerConfigFile)
+	if err != nil {
+		panic(err)
+	}
+	ctrlCfg := make(StringMap)
+	err = yaml.Unmarshal(ctrlCfgBytes, &ctrlCfg)
+	if err != nil {
+		panic(err)
+	}
+	advertiseAddress := stringOrNil(ctrlCfg.mapFromKey("ctrl").mapFromKey("options")["advertiseAddress"])
+	host := fromString(advertiseAddress)
+	anyFailure := host.testPort("controller advertise address")
+
+	web := ctrlCfg.listFromKey("web")
+
+	log.Infof("verifying %d web entries", len(web))
+	for _, item := range web {
```

Comment:
This is fine for iteration one, but I'd rather see us rework the config parsing so it can be reused.

This code will silently break if the configuration files are reworked.